lib/config.js

```diff
@@ -40,7 +49,128 @@ module.exports.getConfig = async function getConfig({
     config.replacers = []
   }
 
-  config['sort-direction'] = validateSortDirection(config['sort-direction'])
+  if (error) throw error
 
   return config
 }
+
+const category = Joi.object()
+  .keys({
+    title: Joi.string().required(),
+    label: Joi.array()
+      .items(Joi.string())
+      .single(),
+    labels: Joi.array()
+      .items(Joi.string())
+      .single()
+  })
+  .xor('label', 'labels')
+
+const searchReplace = Joi.object().keys({
+  search: Joi.string()
+    .required()
+    .error(new Error('"search" is required and must be a regexp or a string'))
+    .description(
+      'The text to search for in a regex or a string representation'
+    ),
+  replace: Joi.string()
+    .allow('')
+    .required()
+    .description(
+      'Replacement of text, it is possible to reference regex match groups by "$number" like $2 for second match group'
+    )
+})
+
+const fields = {
+  branches: Joi.array()
+    .items(Joi.string())
+    .single()
+    .description(
+      'The branches to listen for configuration updates to `.github/release-drafter.yml` and for merge commits. ' +
+        'Useful if you want to test the app on a pull request branch. Default is the repository’s default branch.'
+    ),
+
+  'change-template': Joi.string().description(
+    'The template to use for each merged pull request.'
+  ),
+
+  'no-changes-template': Joi.string().description(
+    'The template to use for when there’s no changes. Default: `"* No changes"`.'
+  ),
+
+  'version-template': Joi.string().description(
+    'The template to use when calculating the next version number for the release. ' +
+      'Useful for projects that don’t use semantic versioning. ' +
+      'Default: `"$MAJOR.$MINOR.$PATCH"'
+  ),
+
+  'name-template': Joi.string()
+    .allow('')
+    .description('The template for the name of the draft release.'),
+
+  'tag-template': Joi.string()
+    .allow('')
+    .description('The template for the tag of the draft release.'),
+
+  'exclude-labels': Joi.array()
+    .items(Joi.string())
+    .description('Exclude pull requests using labels.'),
+
+  'sort-by': Joi.string().valid(SORT_BY.mergedAt, SORT_BY.title),
+
+  'sort-direction': Joi.string().valid(
+    SORT_DIRECTIONS.ascending,
+    SORT_DIRECTIONS.descending
+  ),
+
+  replacers: Joi.array()
+    .items(searchReplace)
+    .description('Search and replace content in the generated changelog body.'),
+
+  categories: Joi.array()
+    .items(category)
+    .description('Categorize pull requests using labels.'),
+
+  template: Joi.string()
+    .required()
+    .description('The template for the body of the draft release.')
+}
+
+const schema = context => {
```

Comment:
It might be better to move the schema (and related objects) to a separate module to keep this module readable.

Reply:
Fair enough

Reply:
Problem is actually with default config, you run into a circular dependency so. I'll most likely just move default_config to schema